src/transaction/TransactionController.ts

```diff
@@ -79,6 +80,12 @@ export interface Transaction {
   estimatedBaseFee?: string;
 }
 
+export interface GasValues {
+  gasPrice?: string;
```

Comment:
I'm new to this EIP-1559 stuff so maybe this is obvious, but which combination of values do we expect to get passed in here? Assuming some custom gas value is used I mean. e.g. are we expecting _either_ `gasPrice` or _both_ `maxFeePerGas` and `maxPriorityFeePerGas`? Or does it make sense for `maxFeePerGas` to ever be provided without `maxPriorityFeePerGas` (or vice versa)? Or can all three be provided?

Reply:
So, for legacy (networks that don't support EIP-1559) you'd only get `gasPrice` in that case. For the transactions you'd expect _both_ `maxFeePerGas` and `maxPriorityFeePerGas`. So, it doesn't really make sense for `maxFeePerGas` or `maxPriorityFeePerGas` to be provided without the other.

> Or can all three be provided?

I guess the way this is now a consumer could provide all three, but in that case `gasPrice` would be ignored and the `tx` would be treated as EIP-1559.

Reply:
Gotcha, that makes sense.

In that case, maybe a union type would be better here? Something like this:
```
type GasValues = { gasPrice: string; } | { maxFeePerGas: string; maxPriorityFeePerGas: string; }
```

That would ensure that either one or the other must be specified, and that if one of the EIP-1559 properties is given, both are given.